webview-ui/src/components/settings/AutoApproveSettings.tsx

```diff
@@ -242,6 +257,59 @@ export const AutoApproveSettings = ({
 								{t("settings:autoApprove.readOnly.outsideWorkspace.description")}
 							</div>
 						</div>
+						<div>
+							<label className="block font-medium mb-1" data-testid="global-ignore-heading">
+								{t("settings:autoApprove.readOnly.globalIgnore.label")}
+							</label>
+							<div className="text-vscode-descriptionForeground text-sm mt-1">
+								{t("settings:autoApprove.readOnly.globalIgnore.description")}
+							</div>
+						</div>
+
+						<div className="flex gap-2">
+							<Input
+								value={globalIgnoreInput}
+								onChange={(e: any) => setGlobalIgnoreInput(e.target.value)}
+								onKeyDown={(e: any) => {
+									if (e.key === "Enter") {
+										e.preventDefault()
+										handleAddGlobalIgnorePattern()
+									}
+								}}
+								placeholder={t("settings:autoApprove.readOnly.globalIgnore.patternPlaceholder")}
+								className="grow"
+								data-testid="global-ignore-input"
+							/>
+							<Button
+								className="h-8"
+								onClick={handleAddGlobalIgnorePattern}
+								data-testid="add-global-ignore-button">
+								{t("settings:autoApprove.readOnly.globalIgnore.addButton")}
+							</Button>
+						</div>
+
+						<div className="flex flex-wrap gap-2">
+							{(globallyIgnoredFiles ?? []).map((pattern, index) => (
+								<Button
+									key={index}
+									variant="secondary"
+									data-testid={`remove-global-ignore-${index}`}
+									onClick={() => {
+										const newPatterns = (globallyIgnoredFiles ?? []).filter((_, i) => i !== index)
+										setCachedStateField("globallyIgnoredFiles", newPatterns)
+
+										vscode.postMessage({
+											type: "updateSettings",
+											updatedSettings: { globallyIgnoredFiles: newPatterns },
+										})
+									}}>
+									<div className="flex flex-row items-center gap-1">
+										<div>{pattern}</div>
+										<X className="text-foreground scale-75" />
+									</div>
+								</Button>
+							))}
+						</div>
```

Comment:
Can you wrap this + the handler above into its own component to reduce the diff with merges but also to keep the AutoApproveSettings component smaller?